src/app/(main)/layout.tsx

```diff
@@ -30,26 +32,28 @@ export default async function RootLayout({
   const isShowCookie = cookieStore.get("isShowCookie")?.value || true;
   const isConsentCookie = cookieStore.get("isConsentCookie")?.value || false;
 
-  if (typeof window !== "undefined") {
-    const govUk = require("govuk-frontend");
-    govUk.initAll();
-  }
-
   return (
-    <html lang="en">
-      <body>
-        {isShowCookie == true && <CookiesBanner />}
-        {isConsentCookie &&
-          environment !== "development" &&
-          globalContent?.googleAnalytics && (
-            <GoogleAnalytics gaId={globalContent?.googleAnalytics} />
-          )}
-        <Header globalConfig={globalContent} />
-        <Banner globalConfig={globalContent} />
-        <div className="layout-wrap">
+    <>
+      {isShowCookie && <CookiesBanner />}
+      {isConsentCookie &&
+        environment !== "development" &&
+        globalContent?.googleAnalytics && (
+          <GoogleAnalytics gaId={globalContent?.googleAnalytics} />
+        )}
+      <a href="#main" className="govuk-skip-link" data-module="govuk-skip-link">
+        Skip to main content
+      </a>
+      <Header globalConfig={globalContent} />
+      <Banner globalConfig={globalContent} />
+      <main className="layout-wrap" id="main">
+        {process.env.NEXT_PUBLIC_API_MOCKING === "enabled" ? (
+          <MSWInitializer>
+            <Suspense>{children}</Suspense>
+          </MSWInitializer>
+        ) : (
           <Suspense>{children}</Suspense>
-        </div>
-      </body>
-    </html>
+        )}
```

Comment:
Is there a reason `MSWInitializer` only wraps the `<Suspense/>` block and not the headers / cookies etc

test code shouldn't really be part of application code, Usually you would wrap your component inside of your tests. 

as a last resort maybe we can abstract some of this away into a `provider.tsx` file https://github.com/apricot13/govuk-nextjs/blob/main/src/app/providers.tsx

Reply:
Good point. I've instead moved the initialisation away from this wrapper and instead check for the mocking env variable and initialise MSW in the root layout file instead. 

After changing this I found I was having trouble getting MSW to initialise before the app renders and performs a data fetch - this meant that the first render contained live data instead of intercepting the API and returning mock data. It showed me mock data after I refreshed the page but that initial render with live data was happening first. 
There are a few discussions about it in these links:
https://github.com/mswjs/msw/discussions/1049
https://github.com/mswjs/msw/issues/73 

Because registering the Service Worker is an asynchronous operation, this MSW [documentation](https://mswjs.io/docs/integrations/browser) recommends to "defer the rendering of your application until the registration Promise resolves..." The way they've implemented it in their example seems a bit more embedded in the code than we'd probably like. Instead to fix this MSW initialisation timing, one solution I found was to check if mocking is enabled and if we aren't in production, to delay the data fetch so that promise can resolve first. It'll only delay the fetch in those circumstances. I'm trying to think of another workaround as it's still technically in our code but this was suggested on a few of the links that I went through.